pretrain.py

```diff
@@ -21,23 +22,41 @@
 dataset = load_dataset(config.data)
 
 
-### Train the tokenizer & Load the pretrained tokenizer
+### Train the tokenizer & Load the pretrained tokenizer ###
 # ! Multiple Possible Tokenizers
 # TODO: Find the best tokenizer
 
 # You can choose the tokenizer type, default is bpe
 tokenizer = train_tokenizer(dataset=dataset, tokenization_type=config.tokenizer, vocab_size=config.vocab_size)
 
-### TODO: Tokenize the dataset
+### Setup Model ###
 
-### TODO: Init the config and the model
-
-config = ...
-model = ...
-data_collator = ...
+esm_config = EsmConfig(
+    vocab_size=config.vocab_size,
+    num_hidden_layers=config.n_layers,
+    hidden_size=config.n_dims,
+    num_attention_heads=config.n_heads,
+)
+model = TFEsmForMaskedLM(
+    config=esm_config,
+)
+data_collator = DataCollatorForLanguageModeling(tokenizer=tokenizer, mlm=True, mlm_probability=0.15)
 
 ### Setup trainer ###
 
-training_args = ...
-trainer = ...
+training_args = TrainingArguments(
+    output_dir="/scratch/output",
```

Comment:
Maybe the `output_dir` should be a parameter? It might be helpful with multiple runs to not overwrite the output from previous runs.